frontend/src/framework/utils/atomUtils.ts

```diff
@@ -0,0 +1,60 @@
+import { DefaultError, QueryClient, QueryKey, QueryObserverOptions, QueryObserverResult } from "@tanstack/query-core";
+
+import { Atom, Getter, atom } from "jotai";
+import { atomWithQuery } from "jotai-tanstack-query";
+import { atomWithReducer } from "jotai/utils";
+
+export function atomWithCompare<Value>(initialValue: Value, areEqual: (prev: Value, next: Value) => boolean) {
```

Comment:
This is actually taken from the Jotai documentation (https://jotai.org/docs/recipes/atom-with-compare). Since it is not really a callback function I could agree on calling it `areEqualFunc`. Agreed?